### DISCUSSION TITLE
If server pool is down, where is the object's new version placed?

### DISCUSSION THREAD

**Discussion by omarkohl** [minio/minio#16840] [Q&A] [answered]
According to the documentation (see the quotes below) an object and all its versions are always stored on the same server pool. What would happen in the following scenario:

* Two server pools P1 and P2
* Versioning is enabled
* Object O1 with versions V1, V2 and V3 is stored on P1
* P1 goes down (all nodes go offline) but P2 remains available
* An S3 client tries to PUT O1 (intending to create V4)

I see two possibilities:
* The operation succeeds and once P1 is back you have the effect that O1 is stored on two different server pools, contradicting the docs
* The operation is prevented by MinIO because it knows that O1 belongs on P1, which is down

If the correct answer is the second scenario, how does that work? Because I thought every object was stored in a self-contained manner without any central metadata.

> MinIO supports adding one or more server pools to existing MinIO deployments for horizontal expansion. When MinIO has multiple server pools available, **an individual object always writes to the same server pool**. If one server pool goes down, objects on other pools remain accessible.

(emphasis mine)

> Within a cluster, MinIO always stores each unique object and all versions of that object on the same Server Pool.

> MinIO optimizes storage of objects across available pools by writing new objects (that is, objects with no existing versions) to the server pool with the most free space compared total amount of free space on all available server pools.

Source: https://min.io/docs/minio/linux/operations/concepts.html

**Comment by harshavardhana** [comment, marked as answer]
An entire pool can never be down, we do not allow I/O in such situations.

All pools must have write quorum to allow writes.

**Reply by omarkohl** [reply]
Thanks for the clarification! Doesn't that contradict the docs?

> When MinIO has multiple server pools available, an individual object always writes to the same server pool. **If one server pool goes down, objects on other pools remain accessible**.

(emphasis mine)
Source: https://min.io/docs/minio/linux/operations/concepts.html

> A server pool represents its own failure domain and the loss of one or more disks or nodes in that pool does not affect the availability of other pools in the deployment.

Source: https://blog.min.io/server-pools-streamline-storage-operations/

**Reply by harshavardhana** [reply]
> A server pool represents its own failure domain and the loss of one or more disks or nodes in that pool does not affect the availability of other pools in the deployment.

This means that if you maintain the quorum within the pool. It means that you maintain quorum while losing nodes on pool1 won't affect (performance on any operations on other pools). 

> When MinIO has multiple server pools available, an individual object always writes to the same server pool. If one server pool goes down, objects on other pools remain accessible.

This must be elaborated, an entire pool can never go down // cc @ravindk89 @djwfyi
